omniscidb/QueryEngine/Compiler/HelperFunctions.cpp

```diff
@@ -89,47 +89,66 @@ void eliminate_dead_self_recursive_funcs(
 
 void optimize_ir(llvm::Function* query_func,
                  llvm::Module* llvm_module,
-                 llvm::legacy::PassManager& pass_manager,
+                 llvm::PassBuilder& PB,
                  const std::unordered_set<llvm::Function*>& live_funcs,
                  const bool is_gpu_smem_used,
                  const CompilationOptions& co) {
   auto timer = DEBUG_TIMER(__func__);
+
+  llvm::LoopAnalysisManager LAM;
+  llvm::FunctionAnalysisManager FAM;
+  llvm::CGSCCAnalysisManager CGAM;
+  llvm::ModuleAnalysisManager MAM;
+  llvm::LoopPassManager LPM;
+  llvm::CGSCCPassManager CGPM;
+
+  PB.registerModuleAnalyses(MAM);
+  PB.registerCGSCCAnalyses(CGAM);
+  PB.registerFunctionAnalyses(FAM);
+  PB.registerLoopAnalyses(LAM);
+  PB.crossRegisterProxies(LAM, FAM, CGAM, MAM);
+
+  llvm::ModulePassManager MPM =
+      PB.buildPerModuleDefaultPipeline(llvm::OptimizationLevel::O3);
```

Comment:
This adds a bunch of passes that are likely not needed and most likely cost us a lot of compile time.

Reply:
```
lmontign@ansatlin08:/localdisk/lmontign/hdk/omniscidb/build$ ./Tests/CodeGeneratorTest
[==========] Running 8 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 8 tests from CodeGeneratorTest
[ RUN      ] CodeGeneratorTest.IntegerConstant
[       OK ] CodeGeneratorTest.IntegerConstant (186 ms)
[ RUN      ] CodeGeneratorTest.IntegerAdd
[       OK ] CodeGeneratorTest.IntegerAdd (176 ms)
[ RUN      ] CodeGeneratorTest.IntegerColumn
[       OK ] CodeGeneratorTest.IntegerColumn (175 ms)
[ RUN      ] CodeGeneratorTest.IntegerExpr
[       OK ] CodeGeneratorTest.IntegerExpr (177 ms)
[ RUN      ] CodeGeneratorTest.IntegerConstantGPU
[       OK ] CodeGeneratorTest.IntegerConstantGPU (283 ms)
[ RUN      ] CodeGeneratorTest.IntegerAddGPU
[       OK ] CodeGeneratorTest.IntegerAddGPU (576 ms)
[ RUN      ] CodeGeneratorTest.IntegerColumnGPU
[       OK ] CodeGeneratorTest.IntegerColumnGPU (289 ms)
[ RUN      ] CodeGeneratorTest.IntegerExprGPU
[       OK ] CodeGeneratorTest.IntegerExprGPU (314 ms)
[----------] 8 tests from CodeGeneratorTest (2177 ms total)

[----------] Global test environment tear-down
[==========] 8 tests from 1 test suite ran. (2177 ms total)
[  PASSED  ] 8 tests.

```